### ISSUE TITLE
How can I get the callback which means the image was really in diskCache

### ISSUE THREAD

**Issue by Liuyulong84** [facebook/fresco#2165]
I wanner use fresco do something about prefetch，it means without a  SimpleDraweeView target, so it seems that PipelineDraweeControllerBuilder is useless.

**Comment by defHLT** [comment]
1. Example with `inDiskCacheSource` from http://frescolib.org/docs/caching.html should work 
2. Depending of what you are going to do with the reference you have or do not have to call `clone()`. Here are some examples: http://frescolib.org/docs/closeable-references.html . Please be sure to close it.

> but I am worried about whether it affects memory
`clone()` does not copy underlying data but only allocates a single object and increments a counter.

**Comment by Liuyulong84** [comment]
thanks  for explaining `clone()`
But For question one，Maybe I didn't describe it in detail enough.
I read the link you sent, And The code here:
`    public static void cacheImageAndCallBackTimely(){

        final String imageUrl = "xxxxxx";
        if (TextUtils.isEmpty(imageUrl)) {
            return;
        }

        ImagePipeline imagePipeline = Fresco.getImagePipeline();
        Uri uri = Uri.parse(imageUrl);

        DataSource<Boolean> inDiskCacheSource = imagePipeline.isInDiskCache(uri);
        DataSubscriber<Boolean> subscriber = new BaseDataSubscriber<Boolean>() {
            @Override
            protected void onNewResultImpl(DataSource<Boolean> dataSource) {
                if (!dataSource.isFinished()) {
                    return;
                }
                boolean isInCache = dataSource.getResult();
                // your code here
                Log.d("LYL",isInCache + "");
                // always false if did no download
            }

            @Override
            protected void onFailureImpl(DataSource<Boolean> dataSource) {

            }
        };
        inDiskCacheSource.subscribe(subscriber, CallerThreadExecutor.getInstance());
    }`
This code ,`inDiskCacheSource()`  ,  The function is a  activity behavior to lookup whether the diskCache exists, 
My problem is  I want to known the callback which stand for file completely exists in disk when downloading  an image .

In other words，I wanner set a listener  monitor the diskCache file ,   How can I known the disk file exists Passively.  Not by looking whether exists.

**Comment by defHLT** [comment]
On second thought, I don't think disk cache is supposed to be used like you want to. If it's just a single image, can you call ImagePipeline#fetchEncodedImage and save the image to disk manually?

**Comment by pranavrao1** [comment]
This is an issue I am having as well. I am using Fresco primarily as a datastorage device. To store gifs but I need a way to have a listener once it has been downloaded to the diskcache
